Bug: syntax error: failed at position 35 ('-')

if clickhouse db/scheme contains special characte, in our case it is '-', then it fails. 

**Caused by: ru.yandex.clickhouse.except.ClickHouseUnknownException: ClickHouse exception, code: 1002, host: 127.0.0.1, port: 8123; Code: 62. DB::Exception: Syntax error: failed at position 35 ('-'): -db.DATABASECHANGELOGLOCK ON CLUSTER '{cluster}' (ID Int64,LOCKED UInt8,LOCKGRANTED Nullable(DateTime64),LOCKEDBY Nullable(String)) ENGINE ReplicatedMergeTree(. Expected one of: token, Dot, UUID, ON, OpeningRoundBracket, storage definition, ENGINE, PARTITION BY, PRIMARY KEY, ORDER BY, SAMPLE BY, TTL, SETTINGS, AS, COMMENT, INTO OUTFILE, FORMAT, end of query. (SYNTAX_ERROR) (version 22.2.2.1)**

--- a/src/main/java/liquibase/ext/clickhouse/lockservice/ClickHouseLockService.java
+++ b/src/main/java/liquibase/ext/clickhouse/lockservice/ClickHouseLockService.java
@@ -52,7 +52,7 @@ public class ClickHouseLockService extends StandardLockService {
       try {
         String query =
             String.format(
-                "SELECT COUNT(*) FROM %s.%s",
+                "SELECT COUNT(*) FROM `%s`.%s",
                 database.getDefaultSchemaName(), database.getDatabaseChangeLogLockTableName());
         int nbRows = getExecutor().queryForInt(new RawSqlStatement(query));
         isLockTableInitialized = nbRows > 0;
@@ -73,7 +73,7 @@ public class ClickHouseLockService extends StandardLockService {
     try {
       String query =
           String.format(
-              "SELECT ID FROM %s.%s LIMIT 1",
+              "SELECT ID FROM `%s`.%s LIMIT 1",
               database.getDefaultSchemaName(), database.getDatabaseChangeLogLockTableName());
       getExecutor().execute(new RawSqlStatement(query));
       hasTable = true;

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogLockTableClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogLockTableClickHouse.java
@@ -54,7 +54,7 @@ public class CreateDatabaseChangeLogLockTableClickHouse
 
     String createTableQuery =
         String.format(
-            "CREATE TABLE IF NOT EXISTS %s.%s "
+            "CREATE TABLE IF NOT EXISTS `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "("
                 + "ID Int64,"

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogTableClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogTableClickHouse.java
@@ -53,7 +53,7 @@ public class CreateDatabaseChangeLogTableClickHouse extends CreateDatabaseChange
 
     String createTableQuery =
         String.format(
-            "CREATE TABLE IF NOT EXISTS %s.%s "
+            "CREATE TABLE IF NOT EXISTS `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "("
                 + "ID String,"

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/InitializeDatabaseChangeLogLockClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/InitializeDatabaseChangeLogLockClickHouse.java
@@ -52,7 +52,7 @@ public class InitializeDatabaseChangeLogLockClickHouse
 
     String clearDatabaseQuery =
         String.format(
-            "ALTER TABLE %s.%s "
+            "ALTER TABLE `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "DELETE WHERE 1 SETTINGS mutations_sync = 1",
             database.getDefaultSchemaName(),
@@ -60,7 +60,7 @@ public class InitializeDatabaseChangeLogLockClickHouse
 
     String initLockQuery =
         String.format(
-            "INSERT INTO %s.%s (ID, LOCKED) VALUES (1, 0)",
+            "INSERT INTO `%s`.%s (ID, LOCKED) VALUES (1, 0)",
             database.getDefaultSchemaName(), database.getDatabaseChangeLogLockTableName());
 
     return SqlGeneratorUtil.generateSql(database, clearDatabaseQuery, initLockQuery);

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/LockDatabaseChangeLogClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/LockDatabaseChangeLogClickHouse.java
@@ -51,7 +51,7 @@ public class LockDatabaseChangeLogClickHouse extends LockDatabaseChangeLogGenera
     String host = String.format("%s %s (%s)", hostname, hostDescription, hostaddress);
     String lockQuery =
         String.format(
-            "ALTER TABLE %s.%s "
+            "ALTER TABLE `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "UPDATE LOCKED = 1,LOCKEDBY = '%s',LOCKGRANTED = %s WHERE ID = 1 AND LOCKED = 0 SETTINGS mutations_sync = 1",
             database.getDefaultSchemaName(),

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/RemoveChangeSetRanStatusClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/RemoveChangeSetRanStatusClickHouse.java
@@ -52,7 +52,7 @@ public class RemoveChangeSetRanStatusClickHouse extends RemoveChangeSetRanStatus
     ChangeSet changeSet = statement.getChangeSet();
     String unlockQuery =
         String.format(
-            "ALTER TABLE %s.%s "
+            "ALTER TABLE `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "DELETE WHERE ID = '%s' AND AUTHOR = '%s' AND FILENAME = '%s' SETTINGS mutations_sync = 1",
             database.getDefaultSchemaName(),

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/UnlockDatabaseChangelogClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/UnlockDatabaseChangelogClickHouse.java
@@ -50,7 +50,7 @@ public class UnlockDatabaseChangelogClickHouse extends UnlockDatabaseChangeLogGe
 
     String unlockQuery =
         String.format(
-            "ALTER TABLE %s.%s "
+            "ALTER TABLE `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "UPDATE LOCKED = 0,LOCKEDBY = null, LOCKGRANTED = null WHERE ID = 1 AND LOCKED = 1 SETTINGS mutations_sync = 1",
             database.getDefaultSchemaName(),

--- a/src/main/java/liquibase/ext/clickhouse/sqlgenerator/UpdateChangeSetChecksumClickHouse.java
+++ b/src/main/java/liquibase/ext/clickhouse/sqlgenerator/UpdateChangeSetChecksumClickHouse.java
@@ -52,7 +52,7 @@ public class UpdateChangeSetChecksumClickHouse extends UpdateChangeSetChecksumGe
     ChangeSet changeSet = statement.getChangeSet();
     String updateChecksumQuery =
         String.format(
-            "ALTER TABLE %s.%s "
+            "ALTER TABLE `%s`.%s "
                 + SqlGeneratorUtil.generateSqlOnClusterClause(properties)
                 + "UPDATE MD5SUM = '%s' WHERE ID = '%s' AND AUTHOR = '%s' AND FILENAME = '%s' SETTINGS mutations_sync = 1",
             database.getDefaultSchemaName(),

